src/freelist.rs

```diff
@@ -76,6 +76,11 @@ where
         if subtype == Self::type_object_raw(py) {
             if let Some(obj) = <Self as PyClassWithFreeList>::get_free_list(py).pop() {
                 ffi::PyObject_Init(obj, subtype);
+                // Workaround for Python issue 35810; no longer necessary in Python 3.8
+                #[cfg(not(Py_3_8))]
+                if ffi::PyType_HasFeature(subtype, ffi::Py_TPFLAGS_HEAPTYPE) != 0 {
```

Comment:
Is `subtype` always heap type?

Reply:
👍 I think it must be, because all our types are now heap types. Similarly for dealloc.